core/kazoo_bindings/src/kazoo_bindings.erl

```diff
@@ -710,7 +713,9 @@ map_processor_fold(#kz_binding{binding_parts=BParts
                   ,_Routing
                   ,RoutingParts
                   ) ->
-    case matches(BParts, RoutingParts) of
+    case matches(BParts, RoutingParts)
+        orelse matches(RoutingParts, BParts)
```

Comment:
I am saying, never once was kazoo_bindings built to support wildcards in the `routing_key`; why cb_onboard has it is beyond me but it is the **only** place I have seen. The whole purpose of kazoo_bindings is to provide flexible routing of payloads to interested callback modules. So you can bind to `tasks.help.*` and your provided callback will be called when another process calls map/fold with `tasks.help.about` or `tasks.help.foo`.

I think what you're trying to do is query the existing bindings as if it was a database instead of a routing topology. `SELECT * from kazoo_bindings where binding = 'tasks.help.%'`

If that is the case, a new interface must be made to query the existing bindings for matches (where you could do the `matches(RoutingParts, BindParts)` instead.

So I would propose, chers @fenollp and @lazedo, a new set of functions that does binding queries instead of payload routing. I would even consider adding a module to kazoo_bindings, something like `kz_bindings_db` or whatever, to be explicit about using the bindings table as a database vs routing table.

Thoughts?